docs/api/algorand/submit-transaction.mdx

```diff
@@ -13,6 +13,12 @@ Submits a previously issued transaction group that has been signed by the user.
 
 ### Regular response
 
+:::caution
+
+It is very important to check if the transaction has been finalized by checking the (TransactionStatus)[/docs/api/algorand/transaction-status] endpoint after submission to avoid double spending. (Transactions are usually finalized in less than 10 seconds.)
```

Comment:
> to avoid double spending

Well, this is not quite a double spending but rather they would not know that the transaction failed. I would say:

> SubmitTransaction will send the blockchain transaction to the network without waiting for it to be mined. It is important to poll for the transaction status using (TransactionStatus)[/docs/api/algorand/transaction-status] API endpoint after it was submitted, ensuring its status is "confirmed". Normally, transactions are confirmed within 5-10 seconds.

Reply:
My point was that you if you consider a transaction as if it was approved before being finalized, you are enabling users to spend the same funds multiple times (even though you are not double spending in the algorand chain, you are double spending in reference to their service)

Reply:
@ZitRos I updated the warning (straight up copied yours), is there anything left you wanna check before merge?

Reply:
Thanks!